routing/dht/lookup.go

```diff
@@ -40,9 +40,12 @@ func (dht *IpfsDHT) GetClosestPeers(ctx context.Context, key key.Key) (<-chan pe
 		peerset.Add(p)
 	}
 
+	// since the query doesnt actually pass our context down
+	// we have to hack this here. whyrusleeping isnt a huge fan of goprocess
```

Comment:
(meaning that likely the same thing would happen with the contexts -- the derivation of child contexts. the reason for using process there (instead of a bare context) is about rate limiting and proper teardown).